### PR TITLE
add ability to just update manifest

### PR DESCRIPTION
**Pull request by deitch**
**- What I did**

The below are in 2 separate commits, as they stand alone:

* added an option `lkt pkg manifest` that just updates the multi-arch index in the registry, based on the arch-specific images available in the registry
* bumped manifest-tool to latest as of today, which includes fix that allows to select that the index media-type should use OCI index rather than Docker manifest list. This fixes an issue whereby Docker hub would reject updates if the media-type is not OCI index.

**- How I did it**

For the `manifest` command, added a command and parallel files.

For the update to manifest-tools:

1. Submitted a PR that fixed manifest-tools to restore the option to select the media-type
2. updated to latest manifest-tools module
3. updated other dependency
4. crafted 3 `replace` commands in `go.mod` so that the dance of `linuxkit`, `manifest-tools`, `moby/moby`, `docker/docker`, `oras.land/oras-go` all work well together.

**- How to verify it**

Run it, CI.

**- Description for the changelog**

ability to just create manifest in registry
create manifests using OCI index media-type

**- Notes**

Our current process has `lkt pkg push`, which:

1. checks if the images for each arch are available on the registry and, if so, pushes it; else (or if `--force):
5. builds the images locally
6. creates a multi-arch index locally
7. pushes it all up to the registry
8. reviews all known arches in the registry and updates the index

We always had options to force step 2 (do not pull, just rebuild `--force`); to skip or not steps 3&5 (`--manifest=false`); to push or not step 4 (`pkg push` vs `pkg build`).

We do not have an option to _just_ do step 5: look at the registry, and ensure we have an index that includes all known images.

We do it anyways with each `pkg push`, as it is good hygiene. 

This gives us the option to _just_ do step 5. You run:

```
lkt pkg manifest pkg/foo
```

And it will do nothing locally: no builds, no pushes, no pulls. It will just look at the registry, and ensure that the index exists and includes all of the arch-specific images that are there.

### CODE REVIEW

File: src/cmd/linuxkit/pkglib/index.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package pkglib
+
+import (
+	"fmt"
+
+	"github.com/linuxkit/linuxkit/src/cmd/linuxkit/registry"
+	"github.com/linuxkit/linuxkit/src/cmd/linuxkit/util"
+)
+
+func (p Pkg) Index(bos ...BuildOpt) error {
```
Comment (giggsoff) [inline src/cmd/linuxkit/pkglib/index.go:10]: Can you please add comment for this exported function?

Comment (deitch) [reply src/cmd/linuxkit/pkglib/index.go:11]: Done

**Comment by deitch** [comment]
Now we need to wait for CI and someone to approve.

File: src/cmd/linuxkit/go.mod [hunk 2]
```diff
@@ -3,55 +3,72 @@ module github.com/linuxkit/linuxkit/src/cmd/linuxkit
...
 	golang.org/x/sys v0.0.0-20220412211240-33da011f77ad
 	google.golang.org/api v0.57.0
 	gopkg.in/yaml.v2 v2.4.0
 )
 
-replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20220414164044-61404de7df1a+incompatible
+replace (
+	// these are for the delicate dance of docker/docker, moby/moby, moby/buildkit, estesp/manifest-tool, oras.land/oras-go, linuxkit/linuxkit
+	github.com/docker/docker => github.com/moby/moby v20.10.3-0.20220728162118-71cb54cec41e+incompatible
+	github.com/estesp/manifest-tool/v2 => github.com/estesp/manifest-tool/v2 v2.0.6-0.20220728154431-89d791ab7966
```
Comment (giggsoff) [inline src/cmd/linuxkit/go.mod:72]: Do we need `github.com/estesp/manifest-tool/v2` replacement here? It is direct usage, It seems we can define this version in `require` section.

Comment (deitch) [reply src/cmd/linuxkit/go.mod:72]: Unfortunately, we do. go modules try to be too smart by half without it here, and end up changing versions of it and everything else. Every other combination creates conflict. The fact that this works makes me happy to leave good enough alone.

Comment (giggsoff) [reply src/cmd/linuxkit/go.mod:72]: Just checked:
```
sed -i '/v2.0.6-0.20220728154431-89d791ab7966/d' go.mod
go get github.com/estesp/manifest-tool/v2@v2.0.6-0.20220728154431-89d791ab7966
go mod tidy
go mod vendor
make local-build
```

**Comment by deitch** [comment]
Been over a week, CI is green, and no other maintainers objecting. I am merging in.

**Pull request merged**